core/src/main/java/hudson/init/InitMilestone.java

```diff
@@ -82,13 +85,29 @@
      * By this milestone, all programmatically constructed extension point implementations
      * should be added.
      */
-    EXTENSIONS_AUGMENTED("Augmented all extensions"), // TODO nothing attains() this so when does it actually happen?
+    EXTENSIONS_AUGMENTED("Augmented all extensions"),
+
+    /**
+     * By this milestone, all the configurations are loaded from file system
+     */
+    SYSTEM_CONFIG_LOADED("System config loaded"),
+
+    /**
+     * By this milestone, the system configuration is adapted just in case any plugin (CasC might be an example) needs
+     * to update configuration files
+     */
+    SYSTEM_CONFIG_ADAPTED("System config adapted"),
 
     /**
      * By this milestone, all jobs and their build records are loaded from disk.
      */
     JOB_LOADED("Loaded all jobs"),
 
+    /**
+     * By this milestone, any job configuration is adapted or updated just in case any plugin is updated
```

Comment:
The wording here is a little confusing `just in case any plugin is updated` doesn't really state an example that would make it obvious to the reader

Reply:
@res0nance what about this?
```suggestion
     * By this milestone, any job configuration is adapted or updated just in case any plugin needs update former/old configurations
```

Reply:
Should be `needs to update`. Other than that, I think its a lot more clear why this milestone exists 👍